src/test/java/io/camunda/testing/assertions/IncidentAssertTest.java

```diff
@@ -0,0 +1,449 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.camunda.testing.extensions.ZeebeAssertions;
+import io.camunda.testing.util.Utilities.ProcessPackLoopingServiceTask;
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.response.ActivateJobsResponse;
+import io.camunda.zeebe.client.api.response.ActivatedJob;
+import io.camunda.zeebe.client.api.response.ProcessInstanceEvent;
+import io.camunda.zeebe.protocol.record.value.ErrorType;
+import java.util.Collections;
+import java.util.Map;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.StringAssert;
+import org.camunda.community.eze.RecordStreamSource;
+import org.camunda.community.eze.ZeebeEngine;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+@ZeebeAssertions
+class IncidentAssertTest {
+
+  public static final String WRONG_VALUE = "wrong value";
+  public static final String ERROR_CODE = "error";
+  public static final String ERROR_MESSAGE = "error occurred";
+
+  private ZeebeClient client;
+  private ZeebeEngine engine;
+
+  // These tests are for testing assertions as well as examples for users
+  @Nested
+  class HappyPathTests {
+
+    private RecordStreamSource recordStreamSource;
+
+    @Test
+    void testHasErrorType() throws InterruptedException {
+      // given
+      deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode(ERROR_CODE)
+          .errorMessage(ERROR_MESSAGE)
+          .send()
+          .join();
+
+      final IncidentAssert incidentAssert = assertThat(actual).extractLatestIncident();
+
+      // then
+      incidentAssert.hasErrorType(ErrorType.UNHANDLED_ERROR_EVENT);
+    }
+
+    @Test
+    void testHasErrorMessage() throws InterruptedException {
+      // given
+      deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode(ERROR_CODE)
+          .errorMessage(ERROR_MESSAGE)
+          .send()
+          .join();
+
+      final IncidentAssert incidentAssert = assertThat(actual).extractLatestIncident();
+
+      // then
+      incidentAssert.hasErrorMessage(
+          "Expected to throw an error event with the code 'error' with message 'error occurred', but it was not caught. No error events are available in the scope.");
```

Comment:
💭 Should you implement the support for string formatting I mentioned earlier we could use it here for the error code and message.